tests/Pinta.Core.Tests/MathematicsTests.cs

```diff
@@ -35,4 +37,77 @@ public void EuclidGCD_RejectsInvalidInputs (int a, int b)
 		Assert.Throws<ArgumentOutOfRangeException> (() => Mathematics.EuclidGCD (a, b));
 		Assert.Throws<ArgumentOutOfRangeException> (() => Mathematics.EuclidGCD (b, a));
 	}
+
+	[TestCaseSource (nameof (lerp_cases))]
+	public void Lerp_ComputesValues (LerpCase values)
+	{
+		double result = Mathematics.Lerp (
+			values.from,
+			values.to,
+			values.frac);
+
+		Assert.That (result, Is.EqualTo (values.result).Within (1e-10));
+	}
+
+	[TestCaseSource (nameof (inv_lerp_cases))]
+	public void InvLerp_ComputesValues (LerpCase values)
+	{
+		double inverse = Mathematics.InvLerp (
+			values.from,
+			values.to,
+			values.result);
+
+		Assert.That (inverse, Is.EqualTo (values.frac).Within (1e-10));
+	}
+
+	[TestCaseSource (nameof (one_way_lerp_cases))]
+	public void InvLerp_RejectsSame (LerpCase values)
+	{
+		Assert.Throws<ArgumentException> (() => Mathematics.InvLerp (values.from, values.to, values.result));
+	}
+
+	private static readonly IReadOnlyList<TestCaseData> lerp_cases =
+		GenerateRegularLerpCases ()
+		.Concat (GenerateOneWayLerpCases ())
+		.Select (c => new TestCaseData (c))
+		.ToArray ();
+
+	private static readonly IReadOnlyList<TestCaseData> inv_lerp_cases =
+		GenerateRegularLerpCases ()
+		.Select (c => new TestCaseData (c))
+		.ToArray ();
+
+	private static readonly IReadOnlyList<TestCaseData> one_way_lerp_cases =
+		GenerateOneWayLerpCases ()
+		.Select (c => new TestCaseData (c))
+		.ToArray ();
+
+	public readonly record struct LerpCase (
+		double from,
+		double to,
+		double frac,
+		double result);
+
+	// These should succeed (in their own way) in both Lerp and InvLerp
+	private static IEnumerable<LerpCase> GenerateRegularLerpCases ()
+	{
+		yield return new (0.0, 10.0, 0.0, 0.0);
+		yield return new (0.0, 10.0, 1.0, 10.0);
+		yield return new (0.0, 10.0, 1.0, 10.0);
```

Comment:
Is there a reason for the duplicate test case here, or just a copy-paste error?

Reply:
@cameronwhite ot seems to be just a copy-paste error. I just removed the repeated test